crates/bindings-csharp/Runtime/Internal/FFI.cs

```diff
@@ -165,6 +177,26 @@ ref uint buffer_len
     [LibraryImport(StdbNamespace)]
     public static partial CheckedStatus row_iter_bsatn_close(RowIter iter_handle);
 
+    [LibraryImport(StdbNamespace)]
+    public static partial CheckedStatus datastore_insert_bsatn(
+        TableId table_id,
+        Span<byte> row,
+        ref uint row_len
+    );
+
+    [LibraryImport(StdbNamespace)]
+    public static unsafe partial CheckedStatus datastore_delete_by_btree_scan_bsatn(
+        IndexId index_id,
+        byte* prefix,
```

Comment:
Use `byte[]` or `Span` or `ReadonlySpan` for safety. Then you won't need those manual `fixed (...)` wrappers on the consumption side.

Reply:
There's no way to offset the buffer otherwise; this mirrors the rust side doing one allocation for the prefix/rstart/rend and then indexing into it.

Reply:
> no way to offset the buffer otherwise

You should be able to do that on Span with simple range indexing.